studio/components/interfaces/Settings/General/Infrastructure/ProjectUpgradeAlert/ProjectUpgradeAlert.tsx

```diff
@@ -83,44 +87,89 @@ const ProjectUpgradeAlert = () => {
-                                const change = (BREAKING_CHANGES as any)[reason]
-                                if (change !== undefined)
-                                  return (
-                                    <li key={reason}>
-                                      <p className="text-foreground">{change.title}</p>
-                                      <p className="flex items-center space-x-1">
-                                        <span>This update has breaking changes. Read more </span>
-                                        <Link href={change.url}>
-                                          <a className="text-brand opacity-90 flex items-center space-x-1">
-                                            <span>here</span>
-                                            <IconExternalLink size="tiny" strokeWidth={2} />
-                                          </a>
-                                        </Link>
-                                      </p>
-                                    </li>
-                                  )
-                                else return null
-                              })}
-                            </ul>
-                          }
-                        />
+                        <Alert_Shadcn_ variant="destructive" title="Breaking changes">
+                          <IconAlertCircle className="h-4 w-4" strokeWidth={2} />
+                          <AlertTitle_Shadcn_>Breaking changes</AlertTitle_Shadcn_>
+                          <AlertDescription_Shadcn_ className="flex flex-col gap-3">
+                            <p>
+                              Your project will be upgraded across major versions of Postgres. This
+                              may involve breaking changes.
+                            </p>
+
+                            <Link href="https://supabase.com/docs/guides/platform/migrating-and-upgrading-projects#caveats">
+                              <a target="_blank" rel="noreferrer">
+                                <Button size="tiny" type="default">
+                                  Visit our documentation to learn more about breaking changes
+                                </Button>
+                              </a>
+                            </Link>
+                          </AlertDescription_Shadcn_>
+                        </Alert_Shadcn_>
+                      )}
+                      {legacyAuthCustomRoles.length > 0 && (
+                        <Alert_Shadcn_
+                          variant="warning"
+                          title="Custom Postgres roles using md5 authentication have been detected"
+                        >
+                          <IconAlertCircle className="h-4 w-4" strokeWidth={2} />
+                          <AlertTitle_Shadcn_>
+                            Custom Postgres roles using md5 authentication have been detected
+                          </AlertTitle_Shadcn_>
```

Comment:
frontload important info

```suggestion
                          <AlertTitle_Shadcn_>
                            Custom Postgres roles will not work automatically after upgrade
                          </AlertTitle_Shadcn_>
```